### PR TITLE
Randomize lockers, reagent tanks, and atmospherics equipment in Meta maint

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

This PR randomizes nearly all of the lockers, water/fuel tanks, and space heaters in Meta maint.  There are some exceptions that apply:

- Water/fuel tanks that were grouped together
- O2 or Firefighting lockers near airlocks
- Space heaters near the space bridge in the NW section of the map
- Emergency Storage sections
- Stuff that clearly had a theme to it (fridge locker for kitchen, etc.)

---

<details>
<summary>Example 1 (stuff that I deliberately skipped)</summary>

StrongDMM_xSlupVRPB6

</details>

<details>
<summary>Example 2</summary>

StrongDMM_oMSOJMD5Mm

</details>

<details>
<summary>Example 3</summary>

StrongDMM_Jwui9U5FDA

</details>

---

<details>
<summary>Locker weight chance</summary>

```dm
/obj/effect/spawner/random/structure/closet_empty = 10,
/obj/structure/closet/emcloset = 2,
/obj/structure/closet/firecloset = 2,
/obj/structure/closet/toolcloset = 2,
/obj/structure/closet/l3closet = 1,
/obj/structure/closet/radiation = 1,
/obj/structure/closet/bombcloset = 1,
/obj/structure/closet/mini_fridge = 1,
```

</details>

<details>
<summary>Reagent tank weight chance</summary>

```dm
/obj/structure/reagent_dispensers/fueltank = 5,
/obj/structure/reagent_dispensers/watertank = 4,
/obj/structure/reagent_dispensers/watertank/high = 1,
```

</details>

<details>
<summary>Atmospherics equipment weight chance</summary>

```dm
/obj/machinery/space_heater = 8,
/obj/machinery/portable_atmospherics/pump = 1,
/obj/machinery/portable_atmospherics/scrubber = 1,
```

</details>

Right now these spawners are a 100% chance.  I would like them to be 80% like the other maint/space structure spawners like girders, grilles, barricades, etc. but I will wait and see what the maptainers think.

Edit - I also fixed this

StrongDMM_izyKNmEeTX

There shouldn't be any windows peeping into maint.

## Why It's Good For The Game

Randomized stuff is awesome and attempts to make the station feel different each round.  It also prevents powergaming to a small extent.

## Changelog

:cl:
qol: Add randomized lockers, reagent tanks, and atmospherics equipment to Meta maint.
fix: Fix ordnance lab having exposed windows peeping into maint.  These have been replaced with rwalls.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
this ruins the roundstart chaplain water tank blessing meta smh my head my head

**Comment by san7890** [comment]
I'll give some time for the others to weigh in, but fast thoughts would probably be that _it should be 100%_ odds. Having some variability is nice, but I can think of a few potential issues right off the bat with it being at lower odds:

1: Maintenance could just look... bad? Like, bare. This is more of a vanity thing for maps where we have the halls chock-loaded with stuff in order to fill in the empty space or create that sort of cramped feeling (especially with the heavy equipment you feature in this PR).

2: If something _really_ unfortunate happens and we somehow get a station full of 100% fuel tanks, it wouldn't be optimal. I think that having the odds be lower than 100% would also worsen the severity of this (in that the _chance_ to get a water tank doesn't just get completely nixed out should the 20% come into play).

I'm also uncertain about the current weight for empty lockers... what did you use to determine the numbers you chose?

Otherwise, I think I like the concept of having randomized maintenance, but I don't know if I'd outright support it. I'm a fan of spawners, and I think grouping the spawners into equipment for maintenance spawns/randomization is a novel concept... that should be weighed.

**Comment by timothymtorres** [comment]
> 1: Maintenance could just look... bad? Like, bare. This is more of a vanity thing for maps where we have the halls chock-loaded with stuff in order to fill in the empty space or create that sort of cramped feeling (especially with the heavy equipment you feature in this PR).

The reason we put it at 80% is so that any sec officers or greytiders won't always know if an area has been touched.  I've seen countless rounds where officers are searching maint for valids and have the whole place mapped out looking for any abnormalities.
 
> 2: If something _really_ unfortunate happens and we somehow get a station full of 100% fuel tanks, it wouldn't be optimal. I think that having the odds be lower than 100% would also worsen the severity of this (in that the _chance_ to get a water tank doesn't just get completely nixed out should the 20% come into play).

As I said in the PR, tanks that are paired together (water and welder) in maint are untouched.  I only applied a random tank spawner if the tank was alone.  There are 13 total in this PR.  The chance of all of them being welder or water is low. (1 in 8192)

> I'm also uncertain about the current weight for empty lockers... what did you use to determine the numbers you chose?

Most of the lockers in maint are empty, with the most other common ones being emcloset and fireclosets.  Toolclosets are a little more rare.  I added radiation for rad storms, bombclosets for meteor storms, and l3closets (biohazard) for xenos and blobs.  Last the fridge since that one is really rare as well.

> ```dm
> /obj/effect/spawner/random/structure/closet_empty = 10,  (50%)
> /obj/structure/closet/emcloset = 2, (10%)
> /obj/structure/closet/firecloset = 2, (10%)
> /obj/structure/closet/toolcloset = 2, (10%)
> /obj/structure/closet/l3closet = 1, (5%)
> /obj/structure/closet/radiation = 1, (5%)
> /obj/structure/closet/bombcloset = 1, (5%)
> /obj/structure/closet/mini_fridge = 1, (5%)
> ```
 
> Otherwise, I think I like the concept of having randomized maintenance, but I don't know if I'd outright support it. I'm a fan of spawners, and I think grouping the spawners into equipment for maintenance spawns/randomization is a novel concept... that should be weighed.

Pure randomization is just as bad as no randomization.  There is a fine balance between too much and too little.

**Comment by Serxule** [comment]
does this mean that plasma cans can spawn in maints?

**Comment by timothymtorres** [comment]
> does this mean that plasma cans can spawn in maints?

As cool as that sounds, none of the random spawners have that.  There is an atmospherics canister but it has a 80% chance of being air and 20% chance of being pure oxygen.

**Comment by Wallemations** [comment]
If those rainbow grilles are the current mapsprite I am begging you to make a custom rainbow cabinet or something similar just to make it easier to tell what's what for mappers

**Comment by tralezab** [comment]
conflict

**Comment by Son-of-Space** [comment]
I'm worried about this removing the chance to put curated loot in maintenance when needed. We don't want all that stuff to be lying on the ground and it prevents us from keeping some areas of maintenance themed to certain loot.

I also don't think heaters should be randomized. At least for me, I deliberately place them to make sure that there are ways to repair breaches. Players who want to fix hull breaches and use their departments shouldn't be penalized because an engineer couldn't be bothered to go searching half the station for a heater. They also shouldn't be penalized repairing minor hull breaches due to a lack of available equipment.

**Comment by timothymtorres** [comment]
> I'm worried about this removing the chance to put curated loot in maintenance when needed. We don't want all that stuff to be lying on the ground and it prevents us from keeping some areas of maintenance themed to certain loot.

My changes do not place any restrictions on mappers from doing this.  In fact there were quite a few maintenance areas that I left alone since it clearly had a theme going for it.
 
> I also don't think heaters should be randomized. At least for me, I deliberately place them to make sure that there are ways to repair breaches. Players who want to fix hull breaches and use their departments shouldn't be penalized because an engineer couldn't be bothered to go searching half the station for a heater. They also shouldn't be penalized repairing minor hull breaches due to a lack of available equipment.

The space heater spawn chance is:

```dm
/obj/machinery/space_heater = 8, (80%)
/obj/machinery/portable_atmospherics/pump = 1, (10%)
/obj/machinery/portable_atmospherics/scrubber = 1, (10%)
```

I do this for several reasons.  While predominantly space heaters should spawn, there is also a small chance for the other atmos equipment to appear as a replacement.  Scrubbers and pumps are criminally underused and having them spawn in maint should give people more options for dealing with plasma flooding, hull breaches, etc.  This allows people to scrub bad air from fires or gas leaks using scrubbers.  They can also replace the air by using a pump.  

Just like the locker spawner, this does not place restrictions on the mappers.  I left several space heaters untouched by the spawner if it was clearly an area that would have heating issues.  (airlocks, space bridge, emergency storage, etc.)

Tldr; Not _every_ locker/space heater/reagent tank in maintenance was replaced by these spawners.
